File: code/JL_OTA_SDK/otasdk_java/src/main/java/com/jieli/otasdk_java/util/auto_test/OTATestTask.java
package com.jieli.otasdk_java.util.auto_test;

import android.bluetooth.BluetoothDevice;

import com.jieli.jl_bt_ota.constant.StateCode;
import com.jieli.jl_bt_ota.interfaces.BtEventCallback;
import com.jieli.otasdk_java.tool.IOtaContract;

/**
 * @ClassName: OTATestTask
 * @Description: java类作用描述
 * @Author: ZhangHuanMing
 * @CreateDate: 2022/2/21 14:40
 */
public class OTATestTask extends TestTask implements IOtaContract.IOtaView {
    private String mFilePath;
    private IOtaContract.IOtaPresenter mOtaPresenter;

    public OTATestTask(String filePath, IOtaContract.IOtaPresenter presenter, TestTaskFinishListener listener) {
        super(listener);
        mFilePath = filePath;
        mOtaPresenter = presenter;
    }

    @Override
    public void start() {
        if (mOtaPresenter.getOtaManager() != null) {

            if (mOtaPresenter.getOtaManager().getDeviceInfo() != null) {
                mOtaPresenter.startOTA(mFilePath);
            } else {
                mOtaPresenter.getOtaManager().registerBluetoothCallback(new BtEventCallback() {
                    @Override
                    public void onConnection(BluetoothDevice device, int status) {
                        if (status == StateCode.CONNECTION_OK) {
                            mOtaPresenter.getOtaManager().unregisterBluetoothCallback(this);
                            mOtaPresenter.startOTA(mFilePath);
                        }
                    }
                });
            }
        }
    }

    private void taskFinish() {
        finishListener.onFinish();
        mOtaPresenter = null;
    }

    @Override
    public boolean isViewShow() {
        return true;
    }

    @Override
    public void onConnection(BluetoothDevice device, int status) {

    }

    @Override
    public void onMandatoryUpgrade() {

    }

    @Override
    public void onOTAStart() {

    }

    @Override
    public void onOTAReconnect(String btAddr) {

    }

    @Override
    public void onOTAProgress(int type, float progress) {

    }

    @Override
    public void onOTAStop() {
        taskFinish();
    }

    @Override
    public void onOTACancel() {

    }

    @Override
    public void onOTAError(int code, String message) {

    }

    @Override
    public void setPresenter(IOtaContract.IOtaPresenter presenter) {

    }


}

File: code/JL_OTA_SDK/otasdk/src/main/java/com/jieli/otasdk/util/auto_test/OTATestTask.kt
package com.jieli.otasdk.util.auto_test

import android.bluetooth.BluetoothDevice
import android.os.Handler
import android.os.Looper
import android.os.Message
import com.jieli.component.thread.ThreadManager
import com.jieli.jl_bt_ota.constant.StateCode
import com.jieli.jl_bt_ota.interfaces.BtEventCallback
import com.jieli.otasdk.base.BasePresenter
import com.jieli.otasdk.tool.IOtaContract
import com.jieli.otasdk.util.TestTask
import com.jieli.otasdk.util.TestTaskFinishListener

/**
 *
 * @ClassName:      OTATestTask
 * @Description:     java类作用描述
 * @Author:         ZhangHuanMing
 * @CreateDate:     2022/2/18 10:55
 */
class OTATestTask(
    private val filePath: String,
    private var otaPresenter: IOtaContract.IOtaPresenter?,
    finishListener: TestTaskFinishListener
) : TestTask(finishListener), IOtaContract.IOtaView {
    override fun start() {
        otaPresenter?.getOtaManager()?.let {
            if (it.deviceInfo != null) {
                otaPresenter?.startOTA(filePath)
            } else {
                it.registerBluetoothCallback(object : BtEventCallback() {
                    override fun onConnection(device: BluetoothDevice?, status: Int) {
                        if (status == StateCode.CONNECTION_OK) {
                            it.unregisterBluetoothCallback(this)
                            otaPresenter?.startOTA(filePath)
                        }
                    }
                })
            }
        }
    }

    private fun taskFinish(): Unit {
        finishListener.onFinish()
        otaPresenter = null
    }

    override fun isViewShow(): Boolean {
        return true
    }

    override fun onConnection(device: BluetoothDevice?, status: Int) {
    }

    override fun onMandatoryUpgrade() {
    }

    override fun onOTAStart() {
    }

    override fun onOTAReconnect(btAddr: String?) {
    }

    override fun onOTAProgress(type: Int, progress: Float) {
    }

    override fun onOTAStop() {//升级成功
        taskFinish()
    }

    override fun onOTACancel() {
        //todo  停止任务
    }

    override fun onOTAError(code: Int, message: String) {
        //todo  停止任务
    }

    override fun setPresenter(presenter: IOtaContract.IOtaPresenter) {
    }

}